### PR TITLE
Add test setup using py.test

### LINKED ISSUE #40: Testing Layout
We need to enable testing layout for this project.
Plan is to use pytest

### PR DESCRIPTION
**Pull request by theskumar**
- [x] add instrunctions in READMDE
- [x] add travis integration
- [x] add https://coveralls.io/ integration
- [x] add some basic integration tests
- [x] Add splinter based live/functional testcase for signup flow
- [x] Documentation for those who are new to `py.test` and testing in general.
- [x] make all the test pass :)

Closes #40

### CODE REVIEW

**Comment by theskumar** [comment]
@vnbang2003 @shankisg guys, i've added test setup. Let me know, if anything is not clear. You can write normal Django class based `TestCase` as well. It maybe just me who find function based test more intuitive.

There is a top level `tests/` folder with stub for unit and functional tests to keep all the test easy to find. But if you write a test anywhere with filename `test_*` it will be discovered and run as well.

Do make use of `tests/fixtures` and `tests/factories` whenever you can. I've only added `UserFactory` for now. But add `WorkshopFactory`, `OrganizationFactory` as required.

File: settings/common.py [hunk 1]
```diff
@@ -31,15 +31,12 @@
...
-SITE_URL = os.environ.get('SITE_URL', '').rstrip('/')
-
 # General project information
 # These are available in the template as SITE_INFO.<title>
 SITE_VARIABLES = {
     'site_name': os.environ.get('SITE_NAME', 'PythonExpress'),
     'site_description': '',
-    'site_url': SITE_URL,
-    'footer': 'Copyright &copy; 2015. Python Software Society of India.'
+    'footer': '&copy; 2014-2015 <a target="_blank" href="http://pssi.org.in/">Python Software Society of India'
```
Comment (kracekumar) [inline settings/common.py]: `</a>` is missing

**Comment by kracekumar** [comment]
How should we go about writing tests ? Should we use `selenium` or just tests `views`. We need to definitely have `integration tests, unit tests, functional tests`. Few points should be documented.

**Comment by theskumar** [comment]
Writing the `unit` and `integration` tests are already in place. The `functional` test setup with selenium is not there yet. 

I've not been writing UI testcases with selenium much, I'll have to research a bit with different options available and correctly set it up. With my current experience, `splinter` with `phantomjs` driver is probably what i am looking for. I'm not sure, if js testrunners like karma will be more appropriate. Suggestions?

> Few points should be documented.

Can you please explain this a bit a more.

_PS: I've renamed folder from `tests/functional/` to `tests/integration/` to better represent what it does._

**Comment by theskumar** [comment]
One the issues writing UI functional test, which I can immediately see is running test would be inherently quite slow. I would prefer to write `integration > unit > functional tests`, but I may be totally wrong here.

**Comment by theskumar** [comment]
I got some interaction tests inside `tests/functional`. By default, they run with `Firefox` as webdriver locally and with `phantomjs` on travis. I've also added some basic documentation around the `py.test` setup and different plugin used.

**Comment by vnbang2003** [comment]
@kracekumar @theskumar To start i think we are good but let's open more issue to request more member to help us writing more and more test case for each part of the product.
If we agree Let's merge the code. @kracekumar Can you please do needful as i am traveling.

**Comment by vnbang2003** [comment]
Am merging this . @theskumar Can you please add more issue with more test case plan so other can also pitch in . Let Mark all the issue for milestone v0.2

**Comment by theskumar** [comment]
@vnbang2003 Yeah. That make sense.

**Comment by kracekumar** [comment]
> Few points should be documented.

When a pull request is sent, the sender should be aware what type of tests are required. Example: When a new `service` function is added, unit test is suffice. 

When a new view is added does all three tests are required ? This will be helpful for most of the contributors.

**Pull request merged**